### PR TITLE
Use latest node-template from Substrate master

### PR DESCRIPTION
**Pull request by shawntabrizi**
1) Clone the substrate repo on master branch
2) Run the script: `./.maintain/node-template-release.sh ./node-template.tar.gz`
3) Extract the `node-template.tar.gz` file
4) Delete all the old node template files
5) Place all files from the extract zip and commit

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,22 +1,14 @@
...
 
 ## Getting Started
 
-Follow the steps below to get started with the Node Template, or get it up and running right from your browser
-in just a few clicks using [Playground](https://playground.substrate.dev/) :hammer_and_wrench:
-
-### Using Nix
-
-Install [nix](https://nixos.org/) and optionally [direnv](https://github.com/direnv/direnv) and [lorri](https://github.com/target/lorri) for a fully plug
-and play experience for setting up the development environment. To get all the correct dependencies activate direnv `direnv allow` and lorri `lorri shell`.
```
Comment (shawntabrizi) [inline README.md:15]: probably these changes to the readme shouldnt all change, but they were not updated on the substrate repo.

needs resolving

File: shell.nix [hunk 2]
```diff
@@ -1,33 +0,0 @@
-let
-  mozillaOverlay =
-    import (builtins.fetchGit {
-      url = "https://github.com/mozilla/nixpkgs-mozilla.git";
-      rev = "57c8084c7ef41366993909c20491e359bbb90f54";
-    });
-  pinned = builtins.fetchGit {
-    # Descriptive name to make the store path easier to identify
```
Comment (shawntabrizi) [inline shell.nix:8]: dunno if this file is useful for someone

**Comment by riusricardo** [comment]
@sacha-l @al3mart Please take a look to this PR.

File: shell.nix (hunk 2 above)
Comment (al3mart) [reply shell.nix:8]: This was discussed [here](https://github.com/substrate-developer-hub/substrate-node-template/pull/170) and included afterwards, so I would say it does for some people :)

**Comment by jimmychu0807** [comment]
- I reverted the `readme` and `shell.nix`
- But on `shell.nix`, who in devhub team has the expertise to maintain it? Otherwise, I prefer not to keep it (removing it tgt with the related `.envrc`).
- Would need another release of node-template when `monthly-2021-07` is tagged in Substrate upstream. This version of node template would just refer to upstream Substrate commit hash.

**Review by jimmychu0807 — APPROVED** [review]

**Comment by NukeManDan** [comment]
We want to use stable tags  for this template that mirror substrate @shawntabrizi - what tag does this use? or is it floating between monthly builds?

File: node/Cargo.toml [hunk 3]
```diff
@@ -8,52 +15,160 @@ license = 'Unlicense'
...
-sp-core = {version = '3.0.0', git = 'https://github.com/paritytech/substrate.git', tag = 'monthly-2021-05'}
-sp-finality-grandpa = {version = '3.0.0', git = 'https://github.com/paritytech/substrate.git', tag = 'monthly-2021-05'}
-sp-inherents = {version = '3.0.0', git = 'https://github.com/paritytech/substrate.git', tag = 'monthly-2021-05'}
-sp-runtime = {version = '3.0.0', git = 'https://github.com/paritytech/substrate.git', tag = 'monthly-2021-05'}
-sp-transaction-pool = {version = '3.0.0', git = 'https://github.com/paritytech/substrate.git', tag = 'monthly-2021-05'}
-substrate-frame-rpc-system = {version = '3.0.0', git = 'https://github.com/paritytech/substrate.git', tag = 'monthly-2021-05'}
-
-# local dependencies
-node-template-runtime = {version = '3.0.0', path = '../runtime'}
+[dependencies.frame-benchmarking]
```
Comment (NukeManDan) [inline node/Cargo.toml:25]: The style of deps here has been requested by many builders to use the one-line style instead like before. Is it possible to change this upstream and the automated script to parse this updated to accommodate?

File: runtime/src/lib.rs [hunk 4]
```diff
@@ -363,10 +372,6 @@ impl_runtime_apis! {
 		) -> sp_inherents::CheckInherentsResult {
 			data.check_extrinsics(&block)
 		}
-
```
Comment (NukeManDan) [inline runtime/src/lib.rs:366]: curious: is this no longer needed? why was it removed upstream?

It is in the https://github.com/substrate-developer-hub/substrate-parachain-template/ commented and likely would also need to be removed

**Pull request merged**